### ISSUE TITLE
MySurveys - Not seeing any results for many questions. 

### ISSUE THREAD

**Issue by gamingchange113** [surveyjs/survey-library#9151]
**Describe the bug**
I know 14+ people got to the end of my survey, but 4 of 13 of the questions are showing as empty in the results. 

**Steps to reproduce**

1. Go to https://surveyjs.io/Results/Survey/866f7cb6-6589-4f6e-8b52-86800b538fbd
2. Click on any format of results
3. Scroll down to the sixth question, until the 24th question. 
4. See that all responses are empty.

**Expected behavior**
Expected to see answers in some of these questions, from all respondents. There are "repeat" questions, since they q's are conditional, but all respondents are required to answer one version of each question (the version they were shown, based on previous answers). However, no responses are showing from Qs 6-24, for any respondent. 

**Screenshots**
NA

**Please complete the following information:**
- Browser: Edge
- Browser version: Unsure
- JS framework/library: Hosted SurveyJS solution 
- SurveyJS version: Unsure 
- Device: PC 

**Additional context**
Again, the questions are conditional, so while there are 19 questions missing data in the results page, there are really four missing questions, from the respondents' point of view. I've detailed them below. 

Q6: Starts, "Ok, imagine you are told ..." This is a checkboxes question (with two checks) 
Q7: Starts, "You chose Option 1" or "You chose Option 2" (based on their response in Q6). This is a single line input question. 
Q8: Starts, "Ok, now imagine you are told ..." Another checkboxes question (two checks) 
Q9: Again starts, "You chose Option 1" or "You chose Option 2" based on response in Q8. Another single line input question. 

Respondents would not have been able to proceed to future questions without answering these ones ... but there is no data for their responses. What could be wrong? Many thanks for the help.

**Comment by gamingchange113** [comment]
Note that I have tried to change the "clear hidden question values" setting on the survey, from the default (which I believe it set to "true") to "when question gets hidden," and subsequently, to "never." However, the issue has persisted.

**Comment by gamingchange113** [comment]
@andrewtelnov or @JaneSjs, could one of you please advise on this? The survey I'm trying to launch is time sensitive. Thank you.

**Comment by tsv2013** [comment]
@gamingchange113 First of all you can take survey yourself and check whether all questions are filled and saved. Have you checked your survey before start?

**Comment by gamingchange113** [comment]
Yes, I have tested it multiple times.

Sent from my iPhone

On Dec 11, 2024, at 5:33 AM, tsv2013 ***@***.***> wrote:

﻿

@gamingchange113<https://github.com/gamingchange113> First of all you can take survey yourself and check whether all questions are filled and saved. Have you checked your survey before start?

**Comment by gamingchange113** [comment]
@JaneSjs I ask sorry to pester, but the launching of my company depends on this survey, and I have already lost two days for a time sensitive survey topic.  I have tried/tested everything I can think of. Could someone please take a look at the link above and help me troubleshoot? I can provide the JSON too if helpful (though I’d prefer not to post the whole thing publicly). Thank you.

**Comment by JaneSjs** [comment]
Hello @gamingchange113,
We may need some time to review your survey and research this issue. Meanwhile, let me emphasize that MySurveys is a free service intended primarily for demonstration purposes. We do not recommend using it for production surveys or collecting sensitive data.

**Comment by gamingchange113** [comment]
@JaneSjs thank you for your reply. How much time will you need? Anything else you need me to provide? 

Are you saying that if I purchase the license this issue will go away?

**Comment by JaneSjs** [comment]
Hi @gamingchange113,
It appears that question responses were not saved under proper names.

For instance, a question name is `"r-b-offer-1-B>4"`. However, in a response, the question answer is saved under the following key: `r-b-offer-1-B\u0026gt;4`. Due to this name mismatch, a Dashboard doesn't visualize question answers.

We generally recommend that you avoid using special symbols and punctuation marks in question names. I'll reach out to our developers to clarify whether it's possible to fix this issue. I'll reply as soon as I get any news to share.

**Comment by JaneSjs** [comment]
@gamingchange113, in the meantime, you can navigate to the Details tab of your survey results page and download user responses in JSON/XLSX format.

**Comment by gamingchange113** [comment]
Ok, thank you for your help Jane.

**Comment by tsv2013** [comment]
I've investigates the case. The thing is that all post data sanitized. That's why any strings looks like `<>` tagrs and script blocks are removed from the posted data. We decided to add question names validation in our online survey builder (**MySurveys - Introduce question name validation to prevent unsupported symbols from being used within form element names** - https://github.com/surveyjs/service/issues/2711) and improve end-user notification about failed posts (MySurveys - **Make sure that posting responses and getting survey JSON is always working** - https://github.com/surveyjs/service/issues/2656 unfortunately in our private repo).
Closing as well as **MySurveys - Not seeing any results for many questions due to question name mismatch which is caused by special symbols within question names** - https://github.com/surveyjs/service/issues/2703
